packages/core/src/shared/errors.ts

```diff
@@ -568,6 +582,16 @@ function hasCode<T>(error: T): error is T & { code: string } {
     return typeof (error as { code?: unknown }).code === 'string'
 }
 
+/**
+ * Returns the identifier the given error.
+ * Depending on the implementation, the identifier may exist on a
+ * different property.
+ */
+export function getErrorId(error: Error): string {
```

Comment:
we have `hasCode` already. let's use consistent jargon unless there is a reason not to

```suggestion
export function getErrorCode(error: Error): string {
```